runtime/syntax/vim.vim

```diff
@@ -19,14 +19,14 @@ syn cluster vimCommentGroup	contains=vimTodo,@Spell
 
 " Special and plugin vim commands {{{2
 syn match   vimCommand contained	"\<z[-+^.=]\="
-syn keyword vimOnlyCommand contained	fix[del] sh[ell] P[rint]
+syn keyword vimOnlyCommand contained fix[del] sh[ell] te[aroff] P[rint]
 syn keyword vimStdPlugin contained	DiffOrig Man N[ext] S TOhtml XMLent XMLns
 
 " Vim-specific options {{{2
-syn keyword vimOnlyOption contained	biosk bioskey cp compatible consk conskey cm cryptmethod edcompatible guipty key macatsui mzq mzquantum osfiletype oft renderoptions rop st shelltype sn shortname tenc termencoding ta textauto tx textmode tf ttyfast ttym ttymouse tbi ttybuiltin wiv weirdinvert
+syn keyword vimOnlyOption contained	biosk bioskey cp compatible consk conskey cm cryptmethod edcompatible guipty key macatsui mzq mzquantum osfiletype oft renderoptions rop st shelltype sn shortname tenc termencoding ta textauto tx textmode ttym ttymouse tbi ttybuiltin wiv weirdinvert
```

Comment:
If you want to remove these it is better to also revert 6f35bcfb07bae585928e03301eb2f19813c3e400. Otherwise these options will not be listed anywhere: the main reason this commit exists is that I found options that are both in vimOnlyOption and vimOption.